File: CS451_inventoryapp/src/com/example/cs451_inventoryapp/InventoryItem.java
package com.example.cs451_inventoryapp;

public class InventoryItem {
	private static int idCount = 1;
	
	private int id;
	public int getId() {
		return id;
	}
	
	private String name;
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	
	private void Setup(String itemName) {
		this.id = idCount;
		idCount++;
		this.name = itemName;
	}
	
	public InventoryItem(String itemName) {
		this.Setup(itemName);
	}
	
	public InventoryItem() {
		this("DefaultInventoryItem");
	}
}

File: CS451_Inventory/InventoryController/InventoryItem.cs
using System;

namespace InventorySystem {
	public class InventoryItem {
		private string name;
		public string Name {
			get { return name; }
			set { name = value; }
		}

		private static int IDCount = 1;
		private int id;
		public int ID {
			get { return id; }
		}

		private void Setup() {
			id = IDCount;
			IDCount++;
		}

		public InventoryItem () {
			Setup();
			name = "InvItem";
		}

		public InventoryItem(string name) {
			Setup();
			this.name = name;
		}
	}
}

